### PR TITLE
Remove retry logic when drop collection with deadline error

### PR DESCRIPTION
**Pull request by DragonDriver**
Signed-off-by: dragondriver <jiquan.long@zilliz.com>

issue: https://github.com/milvus-io/milvus/issues/7306

### CODE REVIEW

File: pymilvus/client/stub.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def inner(self, *args, **kwargs):
     return inner
 
 
-def retry_on_rpc_failure(retry_times=10, wait=1):
+def retry_on_rpc_failure(retry_times=10, wait=1, retry_on_deadline=True):
```
Comment (czs007) [inline pymilvus/client/stub.py:45]: just rename retry_on_deadline to  `enable`, deafult  `enable=True`

Comment (DragonDriver) [reply pymilvus/client/stub.py:45]: so what if we will distinguish the behavior of handling `UNAVAILABLE`?
maybe you mean `enable_on_deadline` and default `enable_on_deadline=True`?

Comment (czs007) [reply pymilvus/client/stub.py:45]: Add it（`enable_on_deadline`） when we really want to distinguish. Anyway, this `enable` to seems reasonable and will be reserved even if we add `enable_on_dealine`.

**Pull request merged**